### PR TITLE
Add a CRI Core Check

### PR DESCRIPTION
**Pull request by mfpierre**
### What does this PR do?

* Create a CRI Util for querying the CRI over gRPC on the socket
* Add settings for configuring CRI socket path / timeout
* Create a core check collecting cpu/mem/disk metrics about CRI containers
* Create a container host metadata collector for the CRI
* Create a CRI connectivity diagnosis

### Motivation

Better integration all the possible CRI in kubernetes (and possibly other orchestrators)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/util/containers/cri.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build linux
```
Comment (xvello) [inline pkg/util/containers/cri.go]: I don't know how many transitive dependencies `k8s.io/kubernetes/pkg/kubelet/apis/cri/runtime/v1alpha2` has, but to be safe, should we add a build flag to exclude it out of the puppy agent?

**Review by xvello — COMMENTED** [review]
Look good! 2 nits for consistency with docker. I'd like to discuss:

  - whether to introduce a build tag
  - what testing strategy we can implement (to avoid repeating our partial docker coverage)

File: pkg/collector/corechecks/containers/cri.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+	if err := yaml.Unmarshal(data, c); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Configure parses the check configuration and init the check
+func (c *CRICheck) Configure(config, initConfig integration.Data) error {
+	return c.instance.Parse(config)
```
Comment (xvello) [inline pkg/collector/corechecks/containers/cri.go]: Can you please add a call to [CommonConfigure](https://github.com/DataDog/datadog-agent/blob/master/pkg/collector/corechecks/checkbase.go#L65)

File: pkg/config/config.go [hunk 3]
```diff
@@ -201,6 +201,11 @@ func init() {
 	BindEnvAndSetDefault("kubernetes_pod_annotations_as_tags", map[string]string{})
 	BindEnvAndSetDefault("kubernetes_node_labels_as_tags", map[string]string{})
 
+	// CRI
+	BindEnvAndSetDefault("cri_socket_path", "")              // empty is disabled
```
Comment (xvello) [inline pkg/config/config.go]: Should we also add them in the config template?

**Review by xvello — COMMENTED** [review]
Looks good !
A couple of last nits

File: pkg/collector/corechecks/containers/cri.go (hunk 2 above)
Comment (mfpierre) [reply pkg/collector/corechecks/containers/cri.go]: Of course 🤦‍♂️

**Review by xvello — APPROVED** [review]
:shipit:

**Pull request merged**